source/extensions/filters/http/ext_proc/mutation_utils.cc

```diff
@@ -32,18 +36,30 @@ void MutationUtils::applyHeaderMutations(
     if (!sh.has_header()) {
       continue;
     }
-    // Make "false" the default. This is logical and matches the ext_authz
-    // filter. However, the router handles this same protobuf and uses "true"
-    // as the default instead.
-    const bool append = PROTOBUF_GET_WRAPPED_OR_DEFAULT(sh, append, false);
-    if (append) {
-      headers.addCopy(LowerCaseString(sh.header().key()), sh.header().value());
-    } else {
-      headers.setCopy(LowerCaseString(sh.header().key()), sh.header().value());
+    if (isSettableHeader(sh.header().key())) {
+      // Make "false" the default. This is logical and matches the ext_authz
+      // filter. However, the router handles this same protobuf and uses "true"
+      // as the default instead.
+      const bool append = PROTOBUF_GET_WRAPPED_OR_DEFAULT(sh, append, false);
+      if (append) {
+        headers.addCopy(LowerCaseString(sh.header().key()), sh.header().value());
+      } else {
+        headers.setCopy(LowerCaseString(sh.header().key()), sh.header().value());
+      }
     }
   }
 }
 
+// Ignore attempts to set certain sensitive headers that can break later processing.
+// We may re-enable some of these after further testing. This logic is specific
+// to the ext_proc filter so it is not shared with HeaderUtils.
+bool MutationUtils::isSettableHeader(absl::string_view key) {
+  return !absl::EqualsIgnoreCase(key, Headers::get().HostLegacy.get()) &&
+         !absl::EqualsIgnoreCase(key, Headers::get().Host.get()) &&
+         !absl::EqualsIgnoreCase(key, Headers::get().Method.get()) &&
+         !absl::StartsWithIgnoreCase(key, "x-envoy");
```

Comment:
The whole `x-envoy-*` prefix is overridable via bootstrap (see `header_prefix`), so you probably want to make use of `ThreadSafeSingleton<Http::PrefixValue>`.

Reply:
Ah hah. Using that mechanism now via the "HeaderValues::prefix" method.